File: codeforce/Round427(Div2)/A_KeyRace.py
s, v1, v2, t1, t2 = map(int, input().split())
if(v1 * s + 2 * t1 < v2 * s + 2 * t2):
    print("First")
elif(v1 * s + 2 * t1 > v2 * s + 2 * t2):
    print("Second")
else:
    print("Friendship")

File: Div427/A_KeyRace.cpp
#include<bits/stdc++.h>
using namespace std;

int s, v1, v2, t1, t2;
int cnt1, cnt2;
void sol()
{
    int start(0);
    cin >> s >> v1 >> v2 >> t1 >> t2;



    if(v1 * s + 2 * t1 < v2 * s + 2 * t2)
    {
        cout << "First";
    }
    else if(v1 * s + 2 * t1 > v2 * s + 2 * t2)
        cout << "Second";
    else cout << "Friendship";
}

int main()
{
    sol();
    return 0;
}
